backend/sprocs.py

```diff
@@ -146,9 +145,105 @@ def get_job_postings_by_category(categories: str, db_filename: str) -> str:
             for row in cursor.fetchall()
         ]
 
-    print(f"getJobPostingsByCategory returned {len(postings)} records.")
+        if (n := len(result)) > 1:
+            raise MultipleRecordsFound(expected=1, actual=n)
+        elif n == 0:
+            return None
+        return result[0]
+
+
+def get_model_data_by_category(categories: str, db_filename: str) -> list:
+    """Gathers model-specific data by job category name
+
+    Args:
+        categories (str): comma-sep string of categories
+        db_filename (str): db for connection
+
+    Returns:
+        list: db records [id, category, job_title, description, tags]
+    """
+    items = [item.strip() for item in categories.split(",")]
+    category_req = ", ".join(f"'{item}'" for item in items)
+
+    query = f"""
+        SELECT
+            p.id,
+            c.name,
+            job_title,
+            description,
```

Comment:
For model, recommend replacing description with preprocessed_description.

Reply:
sprocs for display in the UI ought to use the html cleaned, but not preprocessed description.  This has been adjusted.